File: src/main/java/com/graqr/threshr/model/redsky/product/Facet.java
package com.graqr.threshr.model.redsky.product;

import com.fasterxml.jackson.annotation.JsonProperty;
import io.micronaut.serde.annotation.Serdeable;

@Serdeable
public record Facet(
        @JsonProperty("facet_id")
        String facetId,
        String type,
        @JsonProperty("display_name")
        String displayName,
        boolean expand
) {

}

File: src/main/kotlin/com/graqr/threshr/model/redsky/product/Facet.kt
package com.graqr.threshr.model.redsky.product

import com.fasterxml.jackson.annotation.JsonProperty
import io.micronaut.serde.annotation.Serdeable

@Serdeable
data class Facet(
    @param:JsonProperty("facet_id") val facetId: String,
    val type: String,
    @param:JsonProperty("display_name") val displayName: String,
    val expand: Boolean
) {

}
